gateway_test.go

```diff
@@ -617,3 +641,264 @@ func TestGatewayHandlerProtoHTTPRequestWithAllowedTarget(t *testing.T) {
 
 	testMetricsContainsResult(t, mustGetMetricsFactory(t, target), metricsEventGatewayRequest, metricsResultSuccess)
 }
+
+func TestGatewayHandlerBinaryHttpWithTargetRewrite(t *testing.T) {
+	target := createMockEchoGatewayServer(t)
+
+	handler := http.HandlerFunc(target.gatewayHandler)
+
+	config, err := target.gateway.Config(CURRENT_KEY_ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	client := ohttp.NewDefaultClient(config)
+
+	httpRequest, err := http.NewRequest(http.MethodPost, fmt.Sprintf("http://%s%s", "original-1.example", BINARY_HTTP_GATEWAY_ENDPOINT), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	binaryRequest := ohttp.BinaryRequest(*httpRequest)
+
+	encodedRequest, err := binaryRequest.Marshal()
+	if err != nil {
+		t.Fatal(err)
+	}
+	req, context, err := client.EncapsulateRequest(encodedRequest)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	reqEnc := req.Marshal()
+
+	request, err := http.NewRequest(http.MethodPost, BINARY_HTTP_GATEWAY_ENDPOINT, bytes.NewReader(reqEnc))
+	if err != nil {
+		t.Fatal(err)
+	}
+	request.Header.Add("Content-Type", "message/ohttp-req")
+
+	rr := httptest.NewRecorder()
+	handler.ServeHTTP(rr, request)
+
+	if status := rr.Result().StatusCode; status != http.StatusOK {
+		t.Fatal(fmt.Errorf("Result did not yield %d, got %d instead", http.StatusOK, status))
+	}
+
+	bodyBytes, err := ioutil.ReadAll(rr.Body)
```

Comment:
Consider using `io.ReadAll()`. Apparently `ioutil.ReadAll()` is deprecated as of 1.16.

Here and below.

Reply:
I scrubbed all the existing usages of `ioutil.ReadAll` and `ioutil.NopCloser` from this file, which no longer needs `io/ioutil`. There's others elsewhere in the package that I don't want to mess with in this PR.